tests/lava/proc/io/test_dataloader.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+from typing import List, Tuple
+import unittest
+import numpy as np
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.ports.ports import OutPort
+from lava.magma.core.process.variable import Var
+from lava.magma.core.model.py.ports import PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+
+from lava.proc.io.sink import RingBuffer as ReceiveProcess
+from lava.proc.io.dataloader import State, Spike
+
+from lava.magma.core.run_conditions import RunSteps
+from lava.magma.core.run_configs import RunConfig
+
+
+class TestRunConfig(RunConfig):
+    """Run configuration selects appropriate Conv ProcessModel based on tag:
+    floating point precision or Loihi bit-accurate fixed point precision"""
+    def __init__(self, select_tag: str = 'fixed_pt'):
+        super().__init__(custom_sync_domains=None)
+        self.select_tag = select_tag
+
+    def select(
+        self, _, proc_models: List[PyLoihiProcessModel]
+    ) -> PyLoihiProcessModel:
+        # print(proc_models)
+        for pm in proc_models:
+            if self.select_tag in pm.tags:
+                return pm
+        raise AssertionError('No legal ProcessModel found.')
+
+
+class DummyDataset:
+    def __init__(self, shape: tuple) -> None:
+        self.shape = shape
+
+    def __len__(self) -> int:
+        return 10
+
+    def __getitem__(self, id: int) -> Tuple[np.ndarray, int]:
+        data = np.arange(np.prod(self.shape)).reshape(self.shape) + id
+        data = data % np.prod(self.shape)
+        label = id
+        return data, label
+
+
+class SpikeDataset(DummyDataset):
+    def __getitem__(self, id: int) -> Tuple[np.ndarray, int]:
+        data = np.arange(np.prod(self.shape)).reshape(self.shape[::-1]) + id
+        data = data.transpose(np.arange(len(self.shape))[::-1]) % 13
+        data = data >= 10
+        label = id
+        return data, label
+
+
+class DummyProc(AbstractProcess):
+    def __init__(self, shape: tuple) -> None:
+        super().__init__(shape=shape)
+        self.state = Var(shape=shape, init=np.zeros(shape))
+        self.s_out = OutPort(shape=shape)
+
+
+@implements(proc=DummyProc, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('fixed_pt')
+class PyDummyProc(PyLoihiProcessModel):
+    s_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, int)
+    state: np.ndarray = LavaPyType(np.ndarray, int)
+
+    def run_spk(self) -> None:
+        self.s_out.send(self.state)
+
+
+class TestDataloader(unittest.TestCase):
```

Comment:
I would have created separate TestCases for testing the `State` dataloader and the `Spike` dataloader.